utils/withSettings.ts

```diff
@@ -0,0 +1,95 @@
+import { groq } from "next-sanity";
+import { sanityClient } from "./sanity";
+
+type Meta = {
+  _id: string;
+  _rev: string;
+  _type: string;
+  _createdAt: string;
+  _updatedAt: string;
+};
+
+export enum BlacklistType {
+  FP = "fp",
+  MFP = "mfp",
+}
+
+type APISettings = Meta &
+  Settings & {
+    blacklist: Array<{
+      _key: string;
+      id: number;
+      [BlacklistType.FP]: boolean;
+      [BlacklistType.MFP]: boolean;
+    }>;
+  };
+
+export type Blacklist = number[];
+
+type Blacklists = {
+  [BlacklistType.FP]: Blacklist;
+  [BlacklistType.MFP]: Blacklist;
+};
+
+export type Settings = {
+  fromDate: string;
+  toDate: string;
+  blacklists: Blacklists;
+  isTBD: boolean;
+};
```

Comment:
I see that it has not been done in the rest of the project, so I'll just leave this here as a note - but you might want to investigate `sanity-typegen` (as utilized in https://github.com/abakus-ntnu/fondet), that generates typescript types automatically based on the Sanity schema you utilize:) makes this a bit cleaner and less manual